tensorflow/core/lib/io/record_reader.cc

```diff
@@ -208,22 +208,35 @@ Status RecordReader::ReadRecord(uint64* offset, tstring* record) {
   return Status::OK();
 }
 
-Status RecordReader::SkipRecords(uint64* offset, int num_to_skip) {
+Status RecordReader::SkipRecords(uint64* offset, int num_to_skip,
+                                 int* num_skipped) {
   TF_RETURN_IF_ERROR(PositionInputStream(*offset));
 
   Status s;
   tstring record;
+  *num_skipped = 0;
   for (int i = 0; i < num_to_skip; ++i) {
     s = ReadChecksummed(*offset, sizeof(uint64), &record);
     if (!s.ok()) {
       last_read_failed_ = true;
       return s;
     }
     const uint64 length = core::DecodeFixed64(record.data());
-    input_stream_->SkipNBytes(length + kFooterSize);
+
+    // Skip data
+    s = input_stream_->SkipNBytes(length + kFooterSize);
+    if (!s.ok()) {
+      last_read_failed_ = true;
+      if (errors::IsOutOfRange(s)) {
+        s = errors::DataLoss("truncated record at ", *offset);
```

Comment:
include the original error message as well, in case it has useful information

Reply:
Added.